src/amr/solvers/time_integrator/euler.hpp

```diff
@@ -0,0 +1,92 @@
+#ifndef PHARE_CORE_NUMERICS_TIME_INTEGRATOR_EULER_HPP
+#define PHARE_CORE_NUMERICS_TIME_INTEGRATOR_EULER_HPP
+
+#include "initializer/data_provider.hpp"
+#include "amr/solvers/solver_mhd_model_view.hpp"
+
+namespace PHARE::solver
+{
+template<template<typename> typename FVMethodStrategy, typename MHDModel>
+class Euler
+{
+    using Layout        = typename MHDModel::gridlayout_type;
+    using Dispatchers_t = Dispatchers<Layout>;
+
+    using FVMethod_t             = Dispatchers_t::template FVMethod_t<FVMethodStrategy>;
+    using FiniteVolumeEuler_t    = Dispatchers_t::FiniteVolumeEuler_t;
+    using ConstrainedTransport_t = Dispatchers_t::ConstrainedTransport_t;
+    using Faraday_t              = Dispatchers_t::Faraday_t;
+
+    using ToPrimitiveConverter_t    = Dispatchers_t::ToPrimitiveConverter_t;
+    using ToConservativeConverter_t = Dispatchers_t::ToConservativeConverter_t;
+
+public:
+    Euler(PHARE::initializer::PHAREDict const& dict)
+        : fvm_{dict["fv_method"]}
+        , fv_euler_{dict["fv_euler"]}
+        , to_primitive_{dict["to_primitive"]}
+        , to_conservative_{dict["to_conservative"]}
+    {
+    }
+
+    void operator()(auto layouts, auto& state, auto& statenew, auto& fluxes, auto& bc, auto& level,
+                    double const currentTime, double const newTime)
+    {
+        double const dt = newTime - currentTime;
+
+        to_primitive_(layouts, state);
+
+        bc.fillMomentsGhosts(state, level, newTime);
+
+        std::apply(
+            [&](auto&&... fluxArgs) {
+                fvm_(layouts, state, std::forward<decltype(fluxArgs)>(fluxArgs)...);
+
+                // unecessary if we decide to store both primitive and conservative variables
+                to_conservative_(layouts, state);
+
+                fv_euler_(layouts, state, statenew, dt,
+                          std::forward<decltype(fluxArgs)>(fluxArgs)...);
+            },
+            fluxes);
+
+        auto& B_fx = std::get<2>(fluxes);
```

Comment:
so the idea would be to have a struct where fluxes would rather be named members.
What we discuss is that maybe it would even be so that the "local" (per index) VS "grid" (Fields and vecFields) could be somehow abstracted, so that the struct used here (and coming from the solver) would be the same as your PerIndex, or the grid version could maybe somehow give a local one.